tests/neg/OrType.scala

```diff
@@ -2,7 +2,7 @@ class B(val x: Int)
 class C(val x: Double)
 
 object Test{
-  def bar(x: B | C): Int | Double = x.x
+  def bar(x: B | C): Int | Double = x.x  // error
```

Comment:
To summarize it, the change is on the safe, conservative side. It's a simple way to avoid bugs and keep result types small (no propagation of disjunctions). It requires sometimes manual splitting, hence more verbose code. But it seems that code can always transformed through manual splitting to fit the new model.